src/pam_hbac_utils.c

```diff
@@ -83,12 +83,31 @@ null_cstring_array_size(const char *arr[])
     return nelem;
 }
 
+static bool debug_mode = true;
```

Comment:
Please don't mix code and data, it would be better to move the declaration to the top.

As a general issue, I was thinking whether to use __thread if available (with GCC and clang it would...) because pam applications are often multi-threaded. I mean, it's not a big deal for a variable that is set exactly once and the value in the config file rarely changes, but...